include/vara/Feature/FeatureModel.h

```diff
@@ -240,19 +248,44 @@ class FeatureModel {
     return llvm::make_range(Constraints.begin(), Constraints.end());
   }
 
+  Constraint *addConstraint(std::unique_ptr<Constraint> Constraint) {
+    Constraints.push_back(std::move(Constraint));
+    return Constraints.back().get();
+  }
+
   //===--------------------------------------------------------------------===//
-  // Utility
+  // Relationships
 
-  void view() { ViewGraph(this, "FeatureModel-" + this->getName()); }
+  using relationship_iterator = typename RelationshipContainerTy::iterator;
+  using const_relationship_iterator =
+      typename RelationshipContainerTy::const_iterator;
 
-  [[nodiscard]] Feature *getFeature(llvm::StringRef F) const {
-    auto SearchFeature = Features.find(F);
-    if (SearchFeature != Features.end()) {
-      return SearchFeature->getValue().get();
-    }
-    return nullptr;
+  [[nodiscard]] llvm::iterator_range<relationship_iterator> relationships() {
+    return llvm::make_range(Relationships.begin(), Relationships.end());
+  }
+  [[nodiscard]] llvm::iterator_range<const_relationship_iterator>
+  relationships() const {
+    return llvm::make_range(Relationships.begin(), Relationships.end());
+  }
+
+  Relationship *addRelationship(std::unique_ptr<Relationship> Relationship) {
+    Relationships.push_back(std::move(Relationship));
+    return Relationships.back().get();
+  }
+
+  void removeRelationship(Relationship *R) {
```

Comment:
This one should be fine as, for what I know, removing a relationship cannot create an invalid model from a previously valid one.